### PR TITLE
Work squalene mapping mission service

### PR DESCRIPTION
**Pull request by Squalene**
Creates a mapping mission using the Camera data provided by the drone.
Caveat: need an adapter of 2D meters coordinates to Latlng

### CODE REVIEW

**Review by Anghul — APPROVED** [review]

**Review by loh0um — APPROVED** [review]
General comments: 
- Could create an interface for MissionBuilder because we could have TriangleMissionBuilder for example or other shapes. Could eventually also create a Shape interface for parralelogram and other shapes that we might create in the future.
- Maybe Builder is not the best name for a static class since it refers to the Builder pattern.

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/MappingMissionService.kt [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2021  Drone3D-Team
+ * The license can be found in LICENSE at root of the repository
+ */
+
+package ch.epfl.sdp.drone3d.mission
+
+import ch.epfl.sdp.drone3d.drone.DroneService
+
+class MappingMissionService constructor(val droneService: DroneService) {
```
Comment (GabrielFleischer) [inline app/src/main/java/ch/epfl/sdp/drone3d/mission/MappingMissionService.kt:10]: Add an @Inject to DroneService such that Hilt will know that it should retreive the droneService from its context when constructing this.

**Review by GabrielFleischer — CHANGES_REQUESTED** [review]
You should maybe add a Module to specify to Hilt that this service can be used by activities and other services

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/MappingMissionService.kt (hunk 1 above)
Comment (GabrielFleischer) [reply app/src/main/java/ch/epfl/sdp/drone3d/mission/MappingMissionService.kt:10]: Something like this should work (not 100% sure)

```suggestion
class MappingMissionService constructor(val droneService: DroneService) {

@Module
@InstallIn(SingletonComponent::class)
abstract class MappingMissionModule {
    
    @Singleton
    @Binds
    abstract fun bindMappingMissionService(impl: MappingMissionService): MappingMissionService 
}
```

**Review by Jorque — APPROVED** [review]

File: app/src/main/java/ch/epfl/sdp/drone3d/mission/MappingMissionService.kt (hunk 1 above)
Comment (Squalene) [reply app/src/main/java/ch/epfl/sdp/drone3d/mission/MappingMissionService.kt]: Done

**Review by GabrielFleischer — APPROVED** [review]

**Pull request merged**